packages/core/src/transport/sendWithRetryStrategy.ts

```diff
@@ -79,22 +80,32 @@ function send(
   sendStrategy(payload, (response) => {
     state.bandwidthMonitor.remove(payload)
     if (wasRequestSuccessful(response)) {
-      state.isIntakeAvailable = true
+      state.intakeStatus = IntakeStatus.UP
       onSuccess()
     } else {
-      state.isIntakeAvailable = false
+      // do not consider intake down if another ongoing request could succeed
+      state.intakeStatus =
+        state.bandwidthMonitor.ongoingRequestCount > 0 ? IntakeStatus.ERROR_DETECTED : IntakeStatus.DOWN
       onFailure()
     }
   })
 }
 
+function retryQueuedPayloads(state: RetryState, sendStrategy: SendStrategy) {
+  const pendingPayloads = []
+  while (state.queuedPayloads.size() > 0) {
+    pendingPayloads.push(state.queuedPayloads.dequeue()!)
+  }
+  pendingPayloads.map((payload) => sendWithRetryStrategy(payload, state, sendStrategy))
+}
```

Comment:
`retryQueuedPayloads` is a lot simpler to understand IMO, and given the complexity already present in this file, I think it makes sense to keep it simple. I think it would be really interesting to exploring finite state machines here to manage the complexity. WDYT?

Also, I am curious to know what you think about maybe using [structuredClone](https://developer.mozilla.org/en-US/docs/Web/API/structuredClone)'`transferables` to avoid having to  run `state.queuedPayloads.dequeue`? There might be a performance gain to have here as data is transferred rather than cloned? Plus it would avoid having the double iteration issue you pointed out

Reply:
> I think it would be really interesting to exploring finite state machines here to manage the complexity. WDYT?

IIRC, we have already discussed this topic, decided to investigate it but outside of the current project, right?

> Also, I am curious to know what you think about maybe using [structuredClone](https://developer.mozilla.org/en-US/docs/Web/API/structuredClone)'transferables to avoid having to run state.queuedPayloads.dequeue? There might be a performance gain to have here as data is transferred rather than cloned? Plus it would avoid having the double iteration issue you pointed out

I don't know this API, not sure that this point would need this capability.
It gave me another idea though: [e214c57](https://github.com/DataDog/browser-sdk/pull/1700/commits/e214c571ff37cc73518b553a027f50b7a51fa06f)
Let me know if you have any concerns.

Reply:
> IIRC, we have already discussed this topic, decided to investigate it but outside of the current project, right?

Yes, agreed

> It gave me another idea though: [e214c57](https://github.com/DataDog/browser-sdk/pull/1700/commits/e214c571ff37cc73518b553a027f50b7a51fa06f)

This is a much simpler solution! yes happy to go with that